src/Lorentz/Contracts/BaseDAO/FA2.hs

```diff
@@ -5,29 +5,255 @@ module Lorentz.Contracts.BaseDAO.FA2
    ( transfer
    , balanceOf
    , tokenMetadataRegistry
-   , permissionsDescriptor
    , updateOperators
+   , defaultPermissionsDescriptor
    ) where
 
 import Lorentz
 import Lorentz.Contracts.BaseDAO.Types
 import Lorentz.Contracts.Spec.FA2Interface
 
--- | TODO: Implement missing methods
+defaultPermissionsDescriptor :: PermissionsDescriptor
+defaultPermissionsDescriptor = PermissionsDescriptor
+  { pdOperator = OwnerOrOperatorTransfer
+  , pdReceiver = OwnerNoHook
+  , pdSender = OwnerNoHook
+  , pdCustom = Nothing
+  }
+
 transfer :: Entrypoint TransferParams Storage
-transfer = doNothing
+transfer = do
+  iter transferItem
+  nil; pair
+  where
+    transferItem :: forall s. TransferItem & Storage & s :-> Storage & s
+    transferItem = do
+      swap
+      getField #sAdmin
+      Lorentz.sender
+      eq
+      dig @2
+      getField #tiTxs
+      swap
+      getField #tiFrom
+      swap
+      drop @TransferItem
+      dug @2
+      iter transferOne
+      dropN @2
 
-balanceOf :: Entrypoint BalanceRequestParams Storage
-balanceOf = doNothing
+    transferOne :: forall s. TransferDestination & Bool & Address & Storage & s :-> Bool & Address & Storage & s
+    transferOne = do
+      swap
+      dup
+      if_ nop $ do
+        dug @3
+        dug @3
+        checkSender
+        dig @3
+        dig @3
+      dug @3
+      getField #tdTo
+      swap
+      getField #tdAmount
+      swap
+      getField #tdTokenId
+      swap
+      drop @TransferDestination
+      dup
+      push @Natural 0
+      if IsEq
+      then nop
+      else do
+        dup
+        push @Natural 1
+        if IsEq
+        then failCustom_ #fROZEN_TOKEN_NOT_TRANSFERABLE
```

Comment:
>#fROZEN_TOKEN_NOT_TRANSFERABLE

It seems that this error is not mentioned in the spec.

Reply:
It is however, mentioned in tests

Reply:
Yes, appear to be a missed TODO

https://github.com/tqtezos/baseDAO/blob/master/test/Test/BaseDAO/FA2.hs#L270

I think we can include it in this PR.

Reply:
If some error is missing in our baseDAO specification, please add it there. It says the following:
> The list of erros may be inaccurate and incomplete, it will be updated during the implementation.

It's hard to list all errors in advance.